lib/compiler-singlepass/src/emitter_x64.rs

```diff
@@ -471,154 +475,290 @@ macro_rules! jmp_op {
+                    XMM::XMM6 => dynasm!(emitter ; $ins Rx((dst as u8)), xmm6, DWORD [Rq((base as u8)) + disp]),
+                    XMM::XMM7 => dynasm!(emitter ; $ins Rx((dst as u8)), xmm7, DWORD [Rq((base as u8)) + disp]),
+                    XMM::XMM8 => dynasm!(emitter ; $ins Rx((dst as u8)), xmm8, DWORD [Rq((base as u8)) + disp]),
+                    XMM::XMM9 => dynasm!(emitter ; $ins Rx((dst as u8)), xmm9, DWORD [Rq((base as u8)) + disp]),
+                    XMM::XMM10 => dynasm!(emitter ; $ins Rx((dst as u8)), xmm10, DWORD [Rq((base as u8)) + disp]),
+                    XMM::XMM11 => dynasm!(emitter ; $ins Rx((dst as u8)), xmm11, DWORD [Rq((base as u8)) + disp]),
+                    XMM::XMM12 => dynasm!(emitter ; $ins Rx((dst as u8)), xmm12, DWORD [Rq((base as u8)) + disp]),
+                    XMM::XMM13 => dynasm!(emitter ; $ins Rx((dst as u8)), xmm13, DWORD [Rq((base as u8)) + disp]),
+                    XMM::XMM14 => dynasm!(emitter ; $ins Rx((dst as u8)), xmm14, DWORD [Rq((base as u8)) + disp]),
+                    XMM::XMM15 => dynasm!(emitter ; $ins Rx((dst as u8)), xmm15, DWORD [Rq((base as u8)) + disp]),
+                },
+            }
+        }
+    }
+}
+
+macro_rules! sse_i2f_32_fn {
+    ($ins:ident) => {
+        |emitter: &mut AssemblerX64, precision: Precision, src1: XMM, src2: GPROrMemory, dst: XMM| {
+            match src2 {
+                GPROrMemory::GPR(x) => {
+                    move_src_to_dst(emitter, precision, src1, dst);
+                    dynasm!(emitter; $ins Rx((src1 as u8)), Rd((x as u8)))
                 },
+                GPROrMemory::Memory(base, disp) => {
+                    move_src_to_dst(emitter, precision, src1, dst);
+                    dynasm!(emitter ;  $ins Rx((src1 as u8)), DWORD [Rq((base as u8)) + disp])
+                }
             }
         }
     }
 }
 
 macro_rules! avx_round_fn {
-    ($ins:ident, $name:ident, $mode:expr) => {
-        fn $name(&mut self, src1: XMM, src2: XMMOrMemory, dst: XMM) {
+    ($ins:ident, $mode:expr) => {
+        |emitter: &mut AssemblerX64, src1: XMM, src2: XMMOrMemory, dst: XMM| {
             match src2 {
-                XMMOrMemory::XMM(x) => dynasm!(self ; $ins Rx((dst as u8)), Rx((src1 as u8)), Rx((x as u8)), $mode),
-                XMMOrMemory::Memory(base, disp) => dynasm!(self ; $ins Rx((dst as u8)), Rx((src1 as u8)), [Rq((base as u8)) + disp], $mode),
+                XMMOrMemory::XMM(x) => dynasm!(emitter ; $ins Rx((dst as u8)), Rx((src1 as u8)), Rx((x as u8)), $mode),
+                XMMOrMemory::Memory(base, disp) => dynasm!(emitter ; $ins Rx((dst as u8)), Rx((src1 as u8)), [Rq((base as u8)) + disp], $mode),
             }
         }
     }
 }
 
-impl EmitterX64 for Assembler {
+macro_rules! sse_round_fn {
+    ($ins:ident, $mode:expr) => {
+        |emitter: &mut AssemblerX64, precision: Precision, src1: XMM, src2: XMMOrMemory, dst: XMM| {
+            match src2 {
```

Comment:
`vroundss` and `vroundsd` are always used with src1 == src2.
Here the use of `vround` instead of simple `round` makes no sense, as xmm regs only carry a single Float or Double value.
So, the the `move_src_to_dst` is useless (maybe put an `assert!(src1==src2)` to be safe.